src-tauri/src/main.rs

```diff
@@ -0,0 +1,150 @@
+        "default-terminal",
+        Terminal::pretty_name(&TERMINAL.get().unwrap()[0]),
+    )
+    .unwrap();
+}
+
+#[tauri::command]
+async fn start_download(steam_download: steam::SteamDownload) {
+    let default_terminal = TERMINAL.get().unwrap();
+    let working_dir = env::current_dir().unwrap();
+
+    let terminal_to_use = if steam_download.options().terminal().is_none() { default_terminal.first().unwrap() } else { &Terminal::from_index(&steam_download.options().terminal().unwrap()).unwrap() };
+
+    println!("-------------------------DEBUG INFO------------------------");
+    println!("received these values from frontend:");
+    println!("\t- Username: {}", steam_download.username().as_ref().unwrap_or(&String::from("Not provided")));
+    // println!("\t- Password: {}", steam_download.password().as_ref().unwrap_or(&String::from("Not provided"))); Don't log in prod lol
+    println!("\t- App ID: {}", steam_download.app_id());
+    println!("\t- Depot ID: {}", steam_download.depot_id());
+    println!("\t- Manifest ID: {}", steam_download.manifest_id());
+    println!("\t- Output Path: {}", steam_download.output_path());
+    println!("------------------------DEBUG INFORMATION-----------------");
+    println!("\t- Default terminal: {}", Terminal::pretty_name(&default_terminal[0]));
+    println!("\t- Terminal command: {:?}", terminal_to_use.create_command(&steam_download));
+    println!("\t- Working directory: {}", working_dir.display());
+    println!("----------------------------------------------------------");
+
+    terminal_to_use.create_command(&steam_download).spawn().ok();
+}
+
+/// Downloads the DepotDownloader zip file from the internet based on the OS.
+#[tauri::command]
+async fn download_depotdownloader() {
+    let url = get_depotdownloader_url();
+
+    // Where we store the DepotDownloader zip.
+    let zip_filename = format!("DepotDownloader-v{}-{}.zip", DEPOTDOWNLOADER_VERSION, env::consts::OS);
+    let depotdownloader_zip = Path::new(&zip_filename);
+
+    println!("Downloading DepotDownloader for {} to {}/{}", env::consts::OS, env::current_dir().unwrap().display(), depotdownloader_zip.display());
+
+    match depotdownloader::download_file(url.as_str(), depotdownloader_zip).await {
+        Err(e) => {
+            if e.kind() == io::ErrorKind::AlreadyExists {
+                println!("DepotDownloader already exists. Skipping download.");
+                return;
+            }
+
+            println!("Failed to download DepotDownloader: {}", e);
+            return;
+        },
+        _ => {}
+    }
+
+    println!("Succesfully downloaded DepotDownloader from {}", url);
+
+    depotdownloader::unzip(depotdownloader_zip).unwrap();
+    println!("Succesfully extracted DepotDownloader zip.");
+}
+
+/// Checks internet connectivity using Google
+#[tauri::command]
+async fn internet_connection() -> bool {
+    let client = reqwest::Client::builder().timeout(Duration::from_secs(5)).build().unwrap();
```

Comment:
couldn't you just ping `google.com`? seems simpeler

Reply:
Yeah idk this endpoint is used by Android I think so it should be extremely reliable